### PR TITLE
Fix python 3 support in IDA 7.4

### PR DESCRIPTION
**Pull request by trou**
This version gets the netnode part of BinCAT working on IDA 7.4sp1 with both python 2 and 3.
I haven't tested it against older IDA versions, but it should work on all versions >= 7.0 i guess.

Edit: I remember that `keys()` is broken, I haven't had time to debug it.

### CODE REVIEW

**Comment by RobinDavid** [comment]
Hi all!
The keys() issue is just a infinite recursion. The good keys implementation should be:
```python
    def keys(self):
        return [k for k in self.iterkeys()] # it was keys() here causing recursion
```
Appart from that it works like a charm! @williballenthin have you an idea of when this pull request could be merged ?

**Comment by trou** [comment]
Thanks @RobinDavid, I've updated the pull request (I haven't tested it though)

**Review by williballenthin — APPROVED** [review]
thank you!

**Comment by williballenthin** [comment]
same recursion issue (and fix) for `values` and `items`.

**Comment by williballenthin** [comment]
further py3 fixes here: 18f8558c80cadd67333aa8ecc030f92e65e01f8a

tests adjusted and pass on idapython py3.7

**Comment by williballenthin** [comment]
i dont have a idapython py2.7 installation anymore, so might #yolo release this.

File: netnode/netnode.py [hunk 1]
```diff
@@ -175,16 +185,16 @@ def _strset(self, key, value):
         if len(value) > BLOB_SIZE:
             storekey = self._get_next_slot(STR_KEYS_TAG)
             self._n.setblob(value, storekey, STR_KEYS_TAG)
-            self._n.hashset(key, str(storekey), STR_TO_INT_MAP_TAG)
+            self._n.hashset(key, struct.pack('>I', storekey), STR_TO_INT_MAP_TAG)
```
Comment (williballenthin) [inline netnode/netnode.py:188]: we've just realized that this introduced a breaking change in how we're storing data. this means that .idb files manipulated by ida-netnode prior to the PR are incompatible with the new library version.

will probably have to introduce more code to attempt to detect this scenario and maintain compatibility. then, will release a new major version of the library without the compat mode.

Comment (williballenthin) [reply netnode/netnode.py:188]: alternatively, revert this change and keep the storekey as ASCII-encoded decimal number. then v2.0.x will be considered broken. we can release v3.0.0 with reverted behavior.

Comment (williballenthin) [reply netnode/netnode.py:188]: ultimately, i prefer the packed bytes method, but we have a bunch of existing code and IDBs that this library needs to continue to support. so starting to lean towards a revert.

**Pull request merged**